api/ibc/applications/transfer/v1/tx.pulsar.go

```diff
@@ -0,0 +1,2592 @@
+				return protoiface.UnmarshalOutput{NoUnkeyedLiterals: input.NoUnkeyedLiterals, Flags: input.Flags}, fmt.Errorf("proto: MsgUpdateParamsResponse: wiretype end group for non-group")
+			}
+			if fieldNum <= 0 {
+				return protoiface.UnmarshalOutput{NoUnkeyedLiterals: input.NoUnkeyedLiterals, Flags: input.Flags}, fmt.Errorf("proto: MsgUpdateParamsResponse: illegal tag %d (wire type %d)", fieldNum, wire)
+			}
+			switch fieldNum {
+			default:
+				iNdEx = preIndex
+				skippy, err := runtime.Skip(dAtA[iNdEx:])
+				if err != nil {
+					return protoiface.UnmarshalOutput{NoUnkeyedLiterals: input.NoUnkeyedLiterals, Flags: input.Flags}, err
+				}
+				if (skippy < 0) || (iNdEx+skippy) < 0 {
+					return protoiface.UnmarshalOutput{NoUnkeyedLiterals: input.NoUnkeyedLiterals, Flags: input.Flags}, runtime.ErrInvalidLength
+				}
+				if (iNdEx + skippy) > l {
+					return protoiface.UnmarshalOutput{NoUnkeyedLiterals: input.NoUnkeyedLiterals, Flags: input.Flags}, io.ErrUnexpectedEOF
+				}
+				if !options.DiscardUnknown {
+					x.unknownFields = append(x.unknownFields, dAtA[iNdEx:iNdEx+skippy]...)
+				}
+				iNdEx += skippy
+			}
+		}
+
+		if iNdEx > l {
+			return protoiface.UnmarshalOutput{NoUnkeyedLiterals: input.NoUnkeyedLiterals, Flags: input.Flags}, io.ErrUnexpectedEOF
+		}
+		return protoiface.UnmarshalOutput{NoUnkeyedLiterals: input.NoUnkeyedLiterals, Flags: input.Flags}, nil
+	}
+	return &protoiface.Methods{
+		NoUnkeyedLiterals: struct{}{},
+		Flags:             protoiface.SupportMarshalDeterministic | protoiface.SupportUnmarshalDiscardUnknown,
+		Size:              size,
+		Marshal:           marshal,
+		Unmarshal:         unmarshal,
+		Merge:             nil,
+		CheckInitialized:  nil,
+	}
+}
+
+// Code generated by protoc-gen-go. DO NOT EDIT.
+// versions:
+// 	protoc-gen-go v1.27.0
+// 	protoc        (unknown)
+// source: ibc/applications/transfer/v1/tx.proto
+
+const (
+	// Verify that this generated code is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(20 - protoimpl.MinVersion)
+	// Verify that runtime/protoimpl is sufficiently up-to-date.
+	_ = protoimpl.EnforceVersion(protoimpl.MaxVersion - 20)
+)
+
+// MsgTransfer defines a msg to transfer fungible tokens (i.e Coins) between
+// ICS20 enabled chains. See ICS Spec here:
+// https://github.com/cosmos/ibc/tree/master/spec/app/ics-020-fungible-token-transfer#data-structures
+type MsgTransfer struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
+	// the port on which the packet will be sent
+	SourcePort string `protobuf:"bytes,1,opt,name=source_port,json=sourcePort,proto3" json:"source_port,omitempty"`
+	// the channel by which the packet will be sent
+	SourceChannel string `protobuf:"bytes,2,opt,name=source_channel,json=sourceChannel,proto3" json:"source_channel,omitempty"`
+	// the tokens to be transferred
+	Token *types.Coin `protobuf:"bytes,3,opt,name=token,proto3" json:"token,omitempty"`
```

Comment:
What instructs the code generator to import `Coin` here as:

```
types "github.com/cosmos/cosmos-sdk/types"
```

I see in that in x/gov for example the `InitialDeposit` coin is imported as `v1beta1.Coin` (from the `api` module):

```
v1beta1 "cosmossdk.io/api/cosmos/base/v1beta1"
```

The same is true for other dependent types in ibc protos such as:
- `BaseAccount` used in ica
- `codectypes.Any` used for multiple different fields such as pub keys, client/consensus state in `MsgCreateClient`, etc
- `CommitmentProof` imported from ics23 protos

Due to each of these types not implementing the protov2 api `go build` fails on the api package:
```
value.ProtoReflect undefined (type *"github.com/cosmos/cosmos-sdk/x/auth/types".BaseAccount has no field or method ProtoReflect)
```

Do you guys have any ideas how to get around this? @tac0turtle @julienrbrt 

If we are working towards depinject/appmodule without protov2 then I assume this won't be a problem? Are we better waiting until then to address this, and in the meantime just put try to put the pieces in place for module inputs/outputs etc?

Reply:
You need to add an overwrite in your buf.gen.pulsar.yaml.

Reply:
Example: https://github.com/cosmos/cosmos-sdk/blob/main/x/bank/proto/buf.gen.pulsar.yaml

Reply:
Thanks, will give it a shot!

Reply:
I suspect this should work for sdk types but I'm concerned that ics23 will still not have protov2 methods

Reply:
The `Anys` in cosmos-sdk seem to picking up `anybp` (google import) in the api module, instead of `codectypes.Any`.

I see this diff on main, but on sdk v0.50 release tags codectypes is still used. 

```diff
-    opt: plugins=grpc,Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types
+    opt: plugins=grpc,Mgoogle/protobuf/any.proto=github.com/cosmos/gogoproto/types/any
```

Is there some kind of magic needed to use google anys in the api pkg too? I followed your suggestion for the override and it resolved sdk type imports like coin and base account. Still have `go build` failing in `api` here due to Anys and ics23

Reply:
Yeah, that's on main, you should still use the sdk one for gogoproto. However, this shouldn't be needed for pulsar.
Concerning ics23, yep, you may need to generate it here too.

Why are you generating things with `go-grpc` btw? Only `go-pulsar` is sufficient.